### ISSUE TITLE
web3-utils version is old and causing BigInt issues

### ISSUE THREAD

**Issue by iSuslov** [sc-forks/solidity-coverage#899]
When `solidity-coverage` package is installed it causes issues when using `"web3": "^4.16.0"` 

```ts
import { Web3 } from 'web3';
import { types, Web3ZKsyncL2, ZKsyncPlugin } from 'web3-plugin-zksync';

const web3 = new Web3();
const plugin = new ZKsyncPlugin(Web3ZKsyncL2.initWithDefaultProvider(types.Network.EraTestNode));
```

```console
/Users/user/Projects/bug/node_modules/web3-utils/lib/utils.js:242
    if (!isFinite(value) && !isHexStrict(value)) {
         ^
TypeError: Cannot convert a BigInt value to a number
    at isFinite (<anonymous>)

```

**Comment by cgewecke** [comment]
@iSuslov Thanks for reporting.

Could you show the full stack trace of your error? And provide some information about what package manager you're using and what your dependencies are?

Could you also run `npm list web3-utils` at the root of your project and show me the output, so I can see the different versions of the package in your dependency tree? 

(My sense about this is that solidity-coverage's web3-utils should not be polluting your dependency tree if it's installed normally.)

**Comment by iSuslov** [comment]
@cgewecke I started a zksync tutorial from here: https://docs.zksync.io/zksync-era/guides/zksync-101

It is very easy to reproduce (but don't do that, read to the end):

1. Install `npm install -g zksync-cli`
2. Init a project from template `zksync-cli create zksync-101 --template zksync-101`
3. In this project I installed `web3` and `web3-plugin-zksync` using `npm` and created index.js (not ts for simplicity). Contents of `index.js` are in the original message. 

Running this `index.js` gave me the error above. I tried it on an empty project, and it ran without errors. Then followed a lengthy debugging session dealing with npm dependencies, which eventually revealed that removing `solidity-coverage` made my file run successfully. After some investigation, I discovered the exact reason: your package installs a different version of `web3-utils`.

Here's the tricky part: I don't explicitly use `web3-utils` in my code, but it turns out my code needs it, and needs a specific version. Installing the latest version (`npm i web3-utils`) fixes the problem.

Finally, I believe the issue lies not with `solidity-coverage`, but with `web3-plugin-zksync`. Initially, I suspected npm might be incorrectly resolving `web3-utils` and using the version from solidity coverage. However, I discovered that `web3-plugin-zksync` doesn't even list `web3-utils` as a peer dependency. How frustrating to spend time on it.

Closing this and heading over to unload my frustration in the `web3-plugin-zksync` repo.
